docs/topics/tour/kotlin-tour-intermediate-null-safety.md

```diff
@@ -0,0 +1,658 @@
+* Returns the final value of the `totalLength` variable.
+
+In the reduced version, the example uses the [`.sumOf()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/sum-of.html) extension function and provides a lambda expression that:
+
+* For each item in the list, performs a safe cast to `String` using `as?`.
+* Uses a safe call `?.` to access the `length` property if the call doesn't return a `null` value.
+* Uses the Elvis operator `?:` to return `0` if the safe call returns a `null` value.
+
+## Null values and collections
+
+In Kotlin, working with collections often involves handling `null` values and filtering out unnecessary elements. Kotlin
+has useful functions that you can use to write clean, efficient, and null-safe code when working with lists, sets, maps,
+and other types of collections.
+
+To filter `null` values from a list, use the [`filterNotNull()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/filter-not-null.html) function:
+
+```kotlin
+fun main() {
+    val emails: List<String?> = listOf("alice@example.com", null, "bob@example.com", null, "carol@example.com")
+
+    val validEmails = emails.filterNotNull()
+
+    println(validEmails)
+    // [alice@example.com, bob@example.com, carol@example.com]
+}
+```
+{kotlin-runnable="true" id="kotlin-tour-null-safety-filternotnull"}
+
+If you want to perform filtering of `null` values directly when creating a list, use the [`listOfNotNull()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/list-of-not-null.html) function:
+
+```kotlin
+fun main() {
+    val serverConfig = mapOf(
+        "appConfig.json" to "App Configuration",
+        "dbConfig.json" to "Database Configuration"
+    )
+
+    val requestedFile = "appConfig.json"
+    val configFiles = listOfNotNull(serverConfig[requestedFile])
+
+    println(configFiles)
+    // [App Configuration]
+}
+```
+{kotlin-runnable="true" id="kotlin-tour-null-safety-listofnotnull"}
+
+In both of these examples, if all items are `null` values, an empty list is returned.
+
+Kotlin also provides functions that you can use to find values in collections. If a value isn't found, they return `null`
+values instead of triggering an error:
+
+* [`singleOrNull()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/single-or-null.html), looks for only one item by its exact value. If one doesn't exist or there are multiple items with the same value, returns a `null` value.
+* [`maxOrNull()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/max-or-null.html), finds the highest value. If one doesn't exist, returns a `null` value.
+* [`minOrNull()`](https://kotlinlang.org/api/core/kotlin-stdlib/kotlin.collections/min-or-null.html), finds the lowest value. If one doesn't exist, returns a `null` value.
+
+> The `singleOrNull()`, `maxOrNull()`, and `minOrNull()` functions are designed to be used with collections that **don't**
```

Comment:
Again, I feel like the note here breaks the flow (and makes it look like the example is directly connected to the note, and not to the list), consider moving it lower